src/test/resources/Java21RecordDeconstruction.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package some.pkg;
+
+import java.awt.Point;
+
+public record Java21RecordDeconstruction(Point point) {
+
+	public static int getX(Object obj) {
```

Comment:
I'd prefer not to use tabs... they too easily get mangled or rewritten by text editors.